File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        res = float("inf")
        for i in range(len(blocks) - k + 1):
            pointer = i + k
            j = i
            count = 0
            while j < pointer:
                if blocks[j] == "W":
                    count += 1
                j += 1
            res = min(res, count)

        return res

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.js
/**
 * @param {string} blocks
 * @param {number} k
 * @return {number}
 */
var minimumRecolors = function (blocks, k) {
    let res = Infinity
    for (let i = 0; i < blocks.length - k + 1; i++) {
        let pointer = i + k
        let j = i
        let count = 0
        while (j < pointer) {
            if (blocks[j] == "W") {
                count++
            }
            j++
        }
        res = Math.min(res, count)
    }
    return res
};